intelmq/lib/bot.py

```diff
@@ -24,26 +28,34 @@
 class Bot(object):
 
     def __init__(self, bot_id):
+        self.log_buffer = []
         self.parameters = Parameters()
 
         self.current_message = None
         self.last_message = None
         self.message_counter = 0
         self.error_retries_counter = 0
+        self.source_pipeline = None
+        self.destination_pipeline = None
+        self.logger = None
 
-        self.check_bot_id(bot_id)
-        self.bot_id = bot_id
-
-        self.load_system_configuration()
+        try:
+            self.log_buffer.append(('debug',
+                                    '{} initialized with id {}'
+                                    ''.format(self.__class__.__name__,
+                                              bot_id)))
+            self.check_bot_id(bot_id)
+            self.bot_id = bot_id
+
+            self.load_defaults_configuration()
+            self.load_system_configuration()
+            self.logger = utils.log(self.parameters.logging_path, self.bot_id,
+                                    self.parameters.logging_level)
+        except:
+            self.print_log_buffer()
+            raise
 
-        self.logger = utils.log(
-                                self.parameters.logging_path,
-                                self.bot_id,
-                                self.parameters.logging_level
-                               )
         self.logger.info('Bot is starting')
-
-        self.load_defaults_configuration()
         self.load_runtime_configuration()
```

Comment:
The same as before... I just changed this, the exception is first logged and then passed through (to the caller).